indico/queries/datasets.py

```diff
@@ -552,21 +559,22 @@ def __init__(
         dataset_id: int,
         datafile_ids: List[int],
         wait: bool = True,
+        max_wait_time: Union[int, float] = 5
     ):
         self.dataset_id = dataset_id
         self.datafile_ids = datafile_ids
         self.wait = wait
+        self.max_wait_time = max_wait_time
 
     def requests(self):
         yield _ProcessFiles(self.dataset_id, self.datafile_ids)
-        debouncer = Debouncer()
         yield GetDatasetFileStatus(id=self.dataset_id)
         if self.wait:
             while not all(
                 f.status in ["PROCESSED", "FAILED"] for f in self.previous.files
```

Comment:
dumb question - does self.previous pick up the last `yield` in which case it'd mistakenly pick up the debouncer value instead of getdatasetfilestatus?

Reply:
just tested and it does not. I did not think of that tbh so good thing for me to be aware of going forward thanks